packages/cms-lib/fileManager.js

```diff
@@ -51,6 +58,77 @@ async function uploadFolder(portalId, src, dest, { cwd }) {
   }
 }
 
+/**
+ *
+ * @param {number} portalId
+ * @param {object} file
+ * @param {string} dest
+ * @param {string} folderPath
+ */
+async function fetchFile(portalId, file, dest, folderPath) {
+  const relativePath = `${folderPath}/${file.name}.${file.extension}`;
+  const destPath = convertToUnixPath(path.join(dest, relativePath));
+
+  let writeStream;
+
+  try {
+    await fs.ensureFile(destPath);
+    writeStream = fs.createWriteStream(destPath, { encoding: 'binary' });
+  } catch (err) {
+    console.log(err);
+    // logFsError(err);
+    throw err;
+  }
+
+  await http.getOctetStream(
+    portalId,
+    {
+      baseUrl: file.url,
+      uri: '',
+    },
+    writeStream
+  );
+}
+
+/**
+ *
+ * @param {number} portalId
+ * @param {string} dest
+ * @param {string} folderPath
+ */
+async function getFolderContents(portalId, dest, folderPath) {
+  const files = await getFilesByPath(portalId, folderPath);
+  files.objects.forEach(async file => {
+    fetchFile(portalId, file, dest, folderPath);
+  });
```

Comment:
I think we're going to need to use an old school for loop here. See https://medium.com/dailyjs/the-pitfalls-of-async-await-in-array-loops-cf9cf713bfeb